driver-async/src/main/com/mongodb/async/client/gridfs/GridFS.java

```diff
@@ -0,0 +1,205 @@
+package com.mongodb.async.client.gridfs;
+
+import com.mongodb.async.SingleResultCallback;
+import com.mongodb.async.client.MongoCollection;
+import com.mongodb.async.client.MongoDatabase;
+import com.mongodb.client.model.CreateIndexOptions;
+import com.mongodb.client.result.DeleteResult;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+
+public class GridFS {
```

Comment:
Let's follow the convention of MongoDatabase/Collection and make this an interface.